### DISCUSSION TITLE
Using too many fields in an index results in an index name which is being truncated on the db

### DISCUSSION THREAD

**Discussion by corstian** [JasperFx/marten#3064] [General] [answered]
When this happens Marten will try to drop and regenerate the index every time it starts up. Easily solved by explicitly setting the index name though.

**Comment by gfoidl** [comment, marked as answer]
See also [Postgres Limits on Naming](https://martendb.io/configuration/storeoptions.html#postgres-limits-on-naming). Does configuring it this way solve the problem too?

**Comment by mysticmind** [comment]
There is also option to set an index name (define a shorter index name) as well while defining the index.
